### PR TITLE
Language API docs

### LINKED ISSUE #180: Language API?
I was looking through the DDG RequestHandler code (I was curious about the other kinds of parts of the query you could handle besides the remainder, as mentioned in the [Spice basic tutorial](https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/spice/spice_basic_tutorial.md#define-the-handle-function)). I noticed there's a Language API exposed with the `$lang` variable, similar to the `$loc` one for the Location API.

I was thinking it might be pretty useful for IA developers to know that this feature exists, especially if you're interested in getting non-English language speakers to use DDG.

Would you like me to do a quick writeup of that feature and stick it in a PR?

### PR DESCRIPTION
**Pull request by andrey-p**
Fixes #180.

I think I've covered things reasonably well (including how to test with the Language API), but this is probably more of a WIP at the moment.

I was also wondering whether to include a link to [this document](https://github.com/duckduckgo/duckduckgo/blob/d38c71e6a269ec87441a62e87593a89398182e4a/lib/DDG/Manual/Translation.pod) which seems to describe things pretty well (especially `nplurals`, which baffled me when I saw it for the first time).

I also noticed that the IA referred to in `testing_location_api.md` was incorrect, so I added a fix.

### CODE REVIEW

File: duckduckhack/advanced/language_api.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## Language API
+
+While at the moment DuckDuckGo only accepts Instant Answers in English, it sometimes might be useful to know a user's locale. For instance, it is currently used by the [Translation](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Translate/FromTo.pm) Instant Answer.
```
Comment (moollaza) [inline duckduckhack/advanced/language_api.md]: Would slightly reword:

"Sometimes it's useful to know the user's locale. The [core DDG library]... contained in the `$lang` variable.

`$lang` refers to a [Language object](https://github.com/duckduckgo/duckduckgo/blob/master/lib/DDG/Language.pm) with the following properties:

Example data ...

**Note:** The user's locale does not imply their preferred language it simply provides the locale based on the user's IP. Thus, the locale should not be used to determine the display language for an Instant Answer.
"

Basically remove the first para (the translate Spice is actually dead and not likely to come back) and push the disclaimer to the end and make it a little more noticeable.

File: duckduckhack/advanced/language_api.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+'name_in_english' => 'English of United States',
+'locale'          => 'en_US',
+'flagicon'        => 'us'
+```
+
+When testing interactively with `duckpan`, the locale will always be set to `en_US`.
+
+It is possible to set different locales when running automated tests. Please refer to the [Language API testing guide](https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/testing/testing_language_api.md) for more information.
+
+When your Instant Answer goes live, `$lang` will refer to the user's chosen locale.
```
Comment (moollaza) [inline duckduckhack/advanced/language_api.md]: Can remove this line. Also, it's not based on their "chosen" locale, but their IP's locale.

**Comment by moollaza** [comment]
@andrey-p thanks a lot! This looks great. I made a comment about a modification to the ordering/content for the Language_API.md file -- otherwise everything looks good.

File: duckduckhack/advanced/language_api.md (hunk 2 above)
Comment (andrey-p) [inline duckduckhack/advanced/language_api.md]: > Can remove this line.

@moollaza Just to confirm, you mean line 33?

Comment (moollaza) [inline duckduckhack/advanced/language_api.md]: @andrey-p yup, Line 33. It's explained above that `$lang` refers to their locale.

**Comment by moollaza** [comment]
LGTM :+1:
